client/src/components/createReservation/CreateReservation.tsx

```diff
@@ -203,6 +206,83 @@ const CreateReservation = () => {
     }
   };
 
+  const savePerson = async () => {
+    const errors = validateReservation(newReservation);
+    setErrors({ ...errors });
+
+    if (isReservationValid(errors)) {
```

Comment:
We are trying to avoid many chained ifs because they are hard do read. For this first one you could do : 
```suggestion
    if (!isReservationValid(errors)) {
      updateAlertContext({
        type: "error",
        message: "Please fill required fields",
        isVisible: true,
        isLoading: false,
      });
      return 
    }
```